src/constructors.jl

```diff
@@ -0,0 +1,172 @@
+abstract type AbstractHMCSampler <: AbstractMCMC.AbstractSampler end
+
+##############
+### Custom ###
+##############
+
+"""
+    HMCSampler
+
+A `AbstractMCMC.AbstractSampler` for kernels in AdvancedHMC.jl.
+
+# Fields
+
+$(FIELDS)
+
+# Notes
+
+Note that all the fields have the prefix `initial_` to indicate
+that these will not necessarily correspond to the `kernel`, `metric`,
+and `adaptor` after sampling.
+
+To access the updated fields use the resulting [`HMCState`](@ref).
+"""
+Base.@kwdef struct HMCSampler{
+    K<:AbstractMCMCKernel,
+    M<:AbstractMetric,
+    A<:Adaptation.AbstractAdaptor,
+} <: AbstractHMCSampler
+    "[`AbstractMCMCKernel`](@ref)."
+    kernel::K
+    "[`AbstractMetric`](@ref)."
+    metric::M
+    "[`AbstractAdaptor`](@ref)."
+    adaptor::A
+    "Adaptation steps if any"
+    n_adapts::Int = 0
+end
+
+############
+### NUTS ###
+############
+"""
+    NUTS(n_adapts::Int, δ::Real; max_depth::Int=10, Δ_max::Real=1000, init_ϵ::Real=0)
+
+No-U-Turn Sampler (NUTS) sampler.
+
+# Fields
+
+$(FIELDS)
+
+# Usage:
+
+```julia
+NUTS(n_adapts=1000, δ=0.65)  # Use 1000 adaption steps, and target accept ratio 0.65.
+```
+"""
+struct NUTS{T<:AbstractFloat,I,D} <: AbstractHMCSampler
+    "`n_adapts::Int` : Number of adaptation steps."
+    n_adapts::Int
+    "`δ::Real` : Target acceptance rate for dual averaging."
+    δ::T
+    "`max_depth::Int` : Maximum doubling tree depth."
+    max_depth::Int
+    "`Δ_max::Real` : Maximum divergence during doubling tree."
+    Δ_max::T
+    "`init_ϵ::Real` : Initial step size; 0 means automatically searching using a heuristic procedure."
+    init_ϵ::T
+    "[`AbstractIntegrator`](@ref)."
+    integrator_method::I
+    "[`AbstractMetric`](@ref)."
+    metric_type::D
+end
+
+function NUTS(
+    n_adapts,
+    δ;
+    max_depth = 10,
+    Δ_max = 1000.0,
+    init_ϵ = 0.0,
+    integrator_method = :Leapfrog,
+    metric_type = :DiagEuclideanMetric,
+)
+    return NUTS(n_adapts, δ, max_depth, Δ_max, init_ϵ, integrator_method, metric_type)
+end
+
+###########
+### HMC ###
+###########
+"""
+    HMC(ϵ::Real, n_leapfrog::Int)
+
+Hamiltonian Monte Carlo sampler with static trajectory.
+
+# Fields
+
+$(FIELDS)
+
+# Usage:
+
+```julia
+HMC(init_ϵ=0.05, n_leapfrog=10)
+```
+"""
+struct HMC{T<:AbstractFloat,I,D} <: AbstractHMCSampler
```

Comment:
Same here, just
```suggestion
struct HMC{T<:Real,I,D} <: AbstractHMCSampler
```
?